### PR TITLE
Add stack to x/y encoding channel

### PR DESCRIPTION
**Pull request by kanitw**

### CODE REVIEW

File: src/compile/unit.ts [hunk 1]
```diff
@@ -65,7 +65,6 @@ export class UnitModel extends Model {
     // TODO?: ideally we should use config only inside this constructor
     const config = this._config = this._initConfig(spec.config, parent);
 
-    // FIXME move stacked out of config as it's not really a theme.
     // calculate stack properties
     this._stack = stack(mark, encoding, config.mark.stacked);
```
Comment (kanitw) [inline src/compile/unit.ts]: @domoritz What bugs me a bit is internally stack is a Spec level operation, not an encoding level operation.

Comment (domoritz) [inline src/compile/unit.ts]: I see what you mean but I think conceptually this is where it belongs.

Comment (domoritz) [inline src/compile/unit.ts]: Or at least where people might expect it.

**Comment by domoritz** [comment]
LGTM. Why are we not removing stack from anywhere?

**Comment by kanitw** [comment]
@domoritz  IMO, they can still be a config, but shouldn't be the main way for people to change it.
I wonder if it should be `config.mark.stack` or `config.stack` instead of `config.mark.stacked`.

**Comment by domoritz** [comment]
I'd say remove it from config. In the config is like a theme, I don't see why stack would be in there.

**Comment by kanitw** [comment]
>  I don't see why stack would be in there.

Maybe your project want things to be stacked by default or not.  
For example, CompassQL users might want to enumerate stack and make things not stack unless it's explicitly specified.

**Comment by domoritz** [comment]
Okay. In any case, `config.stack` makes more sense now.

**Comment by kanitw** [comment]
Fixed!

**Comment by domoritz** [comment]
But broke the build.

**Comment by kanitw** [comment]
should be good now.

**Pull request merged**